docs/source/deprecation.md

```diff
@@ -0,0 +1,136 @@
+             END OF USE OF DEPRECATED ENTITIES
+============================================================
+```
+
+#### leapp report example entries
+
+```
+----------------------------------------
+Risk Factor: medium
+Title: Usage of deprecated class "IsolatedActions" at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/libraries/repofileutils.py:38
+Summary: IsolatedActions are deprecated
+Since: 2020-01-02
+Location: /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/libraries/repofileutils.py:38
+Near: def get_parsed_repofiles(context=mounting.NotIsolatedActions(base_dir='/')):
+
+----------------------------------------
+Risk Factor: medium
+Title: Usage of deprecated class "IsolatedActions" at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/scansubscriptionmanagerinfo/libraries/scanrhsm.py:8
+Summary: IsolatedActions are deprecated
+Since: 2020-01-02
+Location: /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/scansubscriptionmanagerinfo/libraries/scanrhsm.py:8
+Near:     context = NotIsolatedActions(base_dir='/')
+
+----------------------------------------
+Risk Factor: medium
+Title: Usage of deprecated function "deprecated_method" at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/deprecationdemo/actor.py:21
+Summary: Deprecated for Demo!
+Since: 2020-06-17
+Location: /usr/share/leapp-repository/repositories/system_upgrade/el7toel8/actors/deprecationdemo/actor.py:21
+Near:         self.deprecated_method()
+
+----------------------------------------
+```
+
+## Usage examples of @deprecated
+
+### Functions
+
+```python
+...
+from leapp.utils.deprecation import deprecated
+
+
+@deprecated(since='2020-06-20', message='This function has been deprecated!')
+def some_deprecated_function(a, b, c):
+    pass
+```
+
+### Models
+
+```python
+...
+from leapp.utils.deprecation import deprecated
+
+
+@deprecated(since='2020-06-20', message='This model has been deprecated!')
+class MyModel(Model):
+    topic = SomeTopic
+    some_field = fields.String()
+```
+
+### Classes
+
+```python
+...
+from leapp.utils.deprecation import deprecated
+
+
+@deprecated(since='2020-06-20', message='This class has been deprecated!')
+class MyClass(object):
+    pass
+
+
+# NOTE: Here we need to offset the stacklevel to get the report of the usage not in the derived class constructor
+# but where the derived class has been created.
+# How many levels you need, you will have to test, it depends on if you use the builtin __init__ methods or not,
+# however it gives you the ability to go up the stack until you reach the position you need to.
+@deprecated(since='2020-06-20', message='This class has been deprecated!', stack_level_offset=1)
+class ABaseClass(object):
+    def __init__(self):
+        pass
+
+
+class ADerivedClass(ABaseClass)
+    def __init__(self):
+        super(ADerivedClass, self).__init__()
+```
```

Comment:
Regarding the documentation, do we plan additional one more specific document for depracation in leapp-repository or this is planned to be the only document? If this is the only document, we should be more specific about the deprecation. E.g. what will happen with deprecated models? Does it mean that during the deprecation time it will be still possible to use the deprecated model? will be actors producing & consuming messages of such models during the 6m period or user will have to immadiately change the code?

Reply:
I find these questions quite strange to be honest. 

"What will happen with deprecated models" => What is deprecated is supposed to be still usable until it is removed.
Deprecation is a heads up, that something will stop being there and supported in future. Otherwise the fact that you deprecate it is useless. 
If this is what you want answered, the I suggest a follow up task with extending this information.

Reply:
The point is, that in such case, it could be useful if we could "drop" the report about deprecated functionality use. I mean, when we need to keep the original functionality still working, the deprecation reports will be print even when customer doesn't care.

Reply:
Resolved by the `@supress_deprecation` decorator.